src/main/java/liquibase/ext/mongodb/statement/InsertOneStatement.java

```diff
@@ -20,59 +20,66 @@
+public class InsertOneStatement extends AbstractRunCommandStatement {
 
-    public static final String COMMAND_NAME = "insertOne";
-
-    private final Document document;
-    private final Document options;
+    public static final String COMMAND_NAME = "insert";
 
     public InsertOneStatement(final String collectionName, final String document, final String options) {
         this(collectionName, orEmptyDocument(document), orEmptyDocument(options));
     }
 
     public InsertOneStatement(final String collectionName, final Document document, final Document options) {
-        super(collectionName);
-        this.document = document;
-        this.options = options;
+        super(BsonUtils.toCommand(COMMAND_NAME, collectionName, combine(document, options)));
     }
 
-    @Override
-    public String getCommandName() {
-        return COMMAND_NAME;
-    }
-
-    @Override
-    public String toJs() {
-        return
-                "db." +
-                        getCollectionName() +
-                        "." +
-                        getCommandName() +
-                        "(" +
-                        ofNullable(document).map(Document::toJson).orElse(null) +
-                        ", " +
-                        ofNullable(options).map(Document::toJson).orElse(null) +
-                        ");";
+    private static Document combine(final Document document, final Document options) {
+        Document combined = new Document("documents", singletonList(document));
+        if (nonNull(options)) {
+            combined.putAll(options);
+        }
+        return combined;
     }
 
+    /**
+     * The server responds with { "ok" : 1 } (success) even when this command fails to insert the document.
+     * The contents of the response is checked to see if the document was actually inserted
+     * For more information see the manual page: https://docs.mongodb.com/manual/reference/command/insert/#output
+     *
+     * @param responseDocument the response document
+     * @throws MongoWriteException containing the code and error message if the document failed to insert
+     */
     @Override
-    public void execute(final MongoConnection connection) {
-            final MongoCollection<Document> collection = connection.getDatabase().getCollection(getCollectionName());
-            collection.insertOne(document);
+    public void checkResponse(Document responseDocument) throws MongoException {
+        if(responseDocument.get("n", Integer.class)==1) return;
+        List<Document> writeErrors = responseDocument.getList("writeErrors", Document.class);
+        if(writeErrors.size()==1) {
+            Document firstError = writeErrors.get(0);
+            int code = firstError.getInteger("code");
+            String message = firstError.getString("errmsg");
+            WriteError error = new WriteError(code, message, new BsonDocument());
+            throw new MongoWriteException(error, null);
+        }
```

Comment:
I really didn't like having to write this code! Other operations will need to similar.
It's certainly worth discussing if this is the way we want to go? Specifically if we kept to using `MongoCollection.insertOne` this type of error handling is not required as the driver does it for us. However this method has a change of signature in newer driver versions (it's no longer `void`) which will force this library and clients to upgrade the driver at the same time.